### ISSUE TITLE
[DEMIS2SORMAS] Handle New Profile: DiagnosticReport.effectiveDateTime

### ISSUE THREAD

**Issue by FredrikSchaeferVitagroup** [hzi-braunschweig/SORMAS-Project#5140]
### Situation Description

This issue is part of #5136.

The new profiles for demis notifications hold the "Erstellungs-/Bearbeitungszeitpunkt " in the DiagnosticReport.effectiveDateTime. According to the description [here](https://simplifier.net/demis/laboratoryreport), the dateTime may be replaced by other values (like just date?). The definition of this value says:

`The time or time-period the observed values are related to. When the subject of the report is a patient, this is usually either the time of the procedure or of specimen collection(s), but very often the source of the date/time is not known, only the date/time itself.`

### Feature Description

To be defined.

### Open questions

- What to do with this information? PathogenTest.testDateTime and Sample.sampleDateTime seem to be already mapped.

**Comment by Jan-Boehme** [comment]
@FredrikSchaeferVitagroup 

I don't really understand this one. I can't find the node DiagnosticReport.effectiveDateTime in the linked documentation.

The description is mentioned in the "Details" tab for the node DiagnosticReport.effective[x] but that will be removed with the new profile according to the "diff" view.

Maybe I am getting something wrong here.

**Comment by FredrikSchaeferVitagroup** [comment]
The DiagnosticReport.effectiveDateTime was presented to me in a web conference by RKI. As the whole DIagnosticReport is a new profile for DEMIS, I am quite certain that it at least was intended to be used.

@Jan-Boehme Maybe we should ask RKI about a clear specification of this attribute?

**Comment by Jan-Boehme** [comment]
@FredrikSchaeferVitagroup 

> Maybe we should ask RKI about a clear specification of this attribute?

I think so too. At the moment the information about that node's existence is not documented anywhere in simplifier.
It would be great if they could add it if it is intended for productive use.

Do you have a contact at RKI where we could place this request?

**Comment by FredrikSchaeferVitagroup** [comment]
I just now wrote an email.

**Comment by FredrikSchaeferVitagroup** [comment]
It was stated from Frauenhofer Fokus that `DiagnosticReport.effectiveDateTime` got removed from the profile. Instead, there now is `DiagnosticReport.issued`. I updated the issue's caption accordingly.

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 will try to get some more information this.

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup I had another look at the DEMIS documentation but I could not find the `DiagnosticReport.effectiveDateTime`, even in the diff. Judging from the description, this should be the `messageDate` and would then also be mapped to the `reportDate` in the pathogen test.

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 As visible in the issue's caption and description, `DiagnosticReport.effectiveDateTime` got removed, and there now instead is `DiagnosticReport.issued`.

Do you want this to be mapped to `PathogenTest.reportDate` for each pathogen test created out of the lab message?
What if the pathogen test already exists in SORMAS and just gets updated? Please consider the scenario 

1. when the pathogen test existing in SORMAS already has a reportDate, and
2. when the pathogen test existing in SORMAS does not already have a reportDate.

**Comment by kwa20** [comment]
> @kwa20 As visible in the issue's caption and description, DiagnosticReport.effectiveDateTime got removed, and there now instead is DiagnosticReport.issued.

Yes, I wasn't sure if it was removed and newly added or actually replaced. Maybe that does not matter though because the outcome is probably the same.

Yes, the `PathogenTest.reportDate` should be mapped to each pathogen created out of the lab message.
1. If the pathogen test already exists in SORMAS and already has a `reportDate`, update the `reportDate` with the latest lab message `DiagnosticReport.issued`
2. If the pathogen test exists in SORMAS and does not have a `reportDate`, fill the field with the `DiagnosticReport.issued`

**Comment by FredrikSchaeferVitagroup** [comment]
This ticket may have something to do with #5714 . After #5714 is finished, users will be presented the values different between the lab message to be processed and the related entities saved in SORMAS. It might make sense to just include the `DiagnosticReport.issued` here.

If we would do that, the report date of a pathogen would not be updated automaticaly in case of 2., but the new value would be presented to the user as a difference. The user would then need to confirm the updated value. @kwa20: Does this make more sense than automatically updating the value without notifying the user about it?

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup Yes, I think validation is a very good idea, also because users will then take note of that change.

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 
The `PathogenTest.reportDate` is already filled with `Bundle.Meta.lastUpdated`.
So the questions remains, what to do with the `DiagnosticReport.issued`?

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 So what shall we do with this ticket? So should we 

- add this field somewhere,
-  just discard this ticket and not map it,
- or move this ticket out of the current sprint and deal with it later?

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup Let's move this ticket out of the current sprint and deal with it later

**Comment by FredrikSchaeferVitagroup** [comment]
@JaquM-HZI As this ticket shall apparently be worked on this sprint, could you please specifiy what shall be done?
